### PR TITLE
Revamp docs directory and add more docs

### LINKED ISSUE #14: Move FAQ and Changelog inside Docs
Should we move FAQ `/faq` and changelog `/changelog` to the docs folder. We can get the native TOC goodies and it's just because the contents is more closely resembles a documentation page than a normal page.

Redirects obviously will be made.

cc @WorldLanguages @GrahamSH-LLK

### LINKED ISSUE #15: Separate "Addon APIs Reference" into multiple pages
I think the content of this page is too long and can be separated into multiple pages. `addon.tab.traps` is separated, so why not all of it?

### LINKED ISSUE #16: Rename some of the docs title
Due to the fact that GitHub does not have an automatic page navigation sidebar, some title of these pages are quite vary. I think it is better to rename this pages (and it's links) for consistency.

Some of the changes that I have in my head are as follows.

- "How to Add an Icon to a Setting Name" ⇾ "Adding an Icon to a Setting (Name)"
- "How to Join the Localization Team" ⇾ "Joining the Localization Team"
- "Addon APIs Reference" ⇾ "Addon APIs"
- "The Addon Manifest (addon.json)" ⇾ "Addon Manifest"

Link changes (if wanted) can be achieved swiftly using the alias feature of Hugo.

### LINKED ISSUE #17: Group more pages on docs
The "Developing" section is quite cluttered because there are quite a lot of pages. I think we can decrease this clutter by grouping some pages that shared the same kind. 

Few examples: "About Persistent Scripts," "About Userscripts," "About Userstyles" could be grouped in the "Addon Types" section, and "Addon APIs Reference," "Design," "Redux," etc. could be grouped in the "Reference" section. Both of these section is below the "Developing" section.

Link changes could be mitigated using the alias feature on Hugo.

### PR DESCRIPTION
**Pull request by Hans5958**
Here's the revamp from me on the docs. I have grouped and rename the pages, so it looks nice and tidy, at least in my opinion. 

The new sections/groupings on the docs are "Addon Types" and "Reference." The "Addon Types" includes the "About..." pages, while the "Reference" includes referential documentation such as "Addon API" and "Addon Manifest." Hopefully this section could be expanded for other things, maybe for the "export settings" thing? This why I also put this outside of the others, because I think the "Developing" section sounds like it is for addon developing, not for the extension as a whole.

I also added a basic "Getting Started" section that I improve from the "Installing Scratch Addons" page, targeted for users. Hopefully we can get more documentations that is actually useful for normal users.

There are also some renamings here and there, some visual updates on some pages, and "Addon API" is now separated into multiple pages.

The preview of this site is located on https://zealous-tereshkova-14b759.netlify.app. You can go to https://zealous-tereshkova-14b759.netlify.app/docs to go directly to the docs.

Redirects/Aliases are also working. This means that the old paths will redirect to the new ones. Here are some examples where I linked the old pages.

- https://zealous-tereshkova-14b759.netlify.app/docs/developing/addon-apis-reference
- https://zealous-tereshkova-14b759.netlify.app/docs/developing/the-addon-manifest-addon.json
- https://zealous-tereshkova-14b759.netlify.app/translate (I really forgot to do this for a while! I apologize!)
- https://zealous-tereshkova-14b759.netlify.app/faq

Suggestions and opinions are open, as always.

Resolves #14 
Resolves #15 
Resolves #16 
Resolves #17
Related to #20 

~~Long write-up later, preview is already up on https://zealous-tereshkova-14b759.netlify.app/docs~~

~~For the record, old paths have been redirected to their new places, For example, https://zealous-tereshkova-14b759.netlify.app/docs/developing/addon-apis-reference~~

### CODE REVIEW

File: content/docs/getting-started/installing.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+title: Installing
+weight: 2
+---
+
+## From extension stores
+
+Scratch Addons is available in both Chrome Web Store and Add-ons for Firefox.
```
Comment (apple502j) [inline content/docs/getting-started/installing.md:8]:
```suggestion
Scratch Addons is available in both Chrome Web Store and addons.mozilla.org.
```
"addons.mozilla.org" is the official name of the extension distribution platform for Firefox.

File: content/docs/getting-started/installing.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+- Chrome Web Store (for Google Chrome, Opera, Brave, Vivaldi, Microsoft Edge (>79), and other Chromium-based browsers)  
+  https://chrome.google.com/webstore/detail/fbeffbjdlemaoicjdapfpikkikjoneco  
+- Add-ons for Firefox (for Mozilla Firefox)
```
Comment (apple502j) [inline content/docs/getting-started/installing.md:12]:
```suggestion
- addons.mozilla.org (for Mozilla Firefox)  
```

File: content/docs/getting-started/installing.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+This is the recommended way to install for development purposes. This assumes you have Git installed.
+
+To download the repository, simply clone `https://github.com/ScratchAddons/ScratchAddons.git`. Here's an example.
+
+```sh
+$ git clone https://github.com/ScratchAddons/ScratchAddons.git
+```
+
+If you have GitHub, you can fork the repository in GitHub beforehand, and then clone the fork instead.
```
Comment (apple502j) [inline content/docs/getting-started/installing.md:31]: Drop this section. Users should always clone upstream repository.

File: content/docs/getting-started/installing.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+$ git clone https://github.com/[your-name]/ScratchAddons.git
+```
+
+### Downloading the zipball
+
+If you don't have Git installed, you can try this method instead.
+
+1. Go to the [repository](https://github.com/ScratchAddons/ScratchAddons) and find the download code button
+
+![Screenshot 2020-09-04 at 6 41 12 PM](https://user-images.githubusercontent.com/61319150/92291572-f17ef080-eede-11ea-85e8-fda961a56074.png)
```
Comment (apple502j) [inline content/docs/getting-started/installing.md:43]: Proper alt text?

File: content/docs/getting-started/installing.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+2. Click it and select "Download ZIP"
+
+![Screenshot 2020-09-04 at 6 43 57 PM](https://user-images.githubusercontent.com/61319150/92291570-f0e65a00-eede-11ea-80ee-f32e340327d9.png)
+
+3. Extract the archive into a folder.
+
+### Installing on Google Chrome
+
+1. Go to the [Extension Management page](chrome://extensions) (`chrome://extensions`).
```
Comment (apple502j) [inline content/docs/getting-started/installing.md:53]: The page is not web-navigatable, and I don't think GFM allows that protocol. Unlinkify.

File: content/docs/getting-started/installing.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+![image](https://user-images.githubusercontent.com/61319150/92291792-d3fe5680-eedf-11ea-9d62-8e02b262f575.png)
+
+4. Select the folder 
+5. The extension should now be loaded. 
+
+### Installing on Mozilla Firefox
+
+<!-- TODO: Add pictures -->
+
+1. Go to the [Debugging page](about://debugging#/runtime/this-firefox) and click "This Firefox" (`about://debugging#/runtime/this-firefox`).
```
Comment (apple502j) [inline content/docs/getting-started/installing.md:70]: The page is not web-navigatable, and I don't think GFM allows that protocol. Unlinkify.

File: content/docs/getting-started/installing.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+2. Click `Load Temporary Add-on...`
```
Comment (apple502j) [inline content/docs/getting-started/installing.md:71]: Either add dots on 2 and 3 or drop on 1 and 4

File: content/docs/getting-started/installing.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+2. Click the toggle next to `Developer mode` to turn on the developer mode. This allows you to install the addon from a file.
+
+![image](https://user-images.githubusercontent.com/61319150/92291715-8550bc80-eedf-11ea-9ec1-6628c5b3fbc2.png)
+
+3. You should see `Load unpacked`. Clicking it will allow you to select a folder to upload.
+
+![image](https://user-images.githubusercontent.com/61319150/92291792-d3fe5680-eedf-11ea-9d62-8e02b262f575.png)
+
+4. Select the folder
```
Comment (apple502j) [inline content/docs/getting-started/installing.md:63]:
```suggestion
4. Select the folder.
```

File: content/docs/getting-started/installing.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+```sh
+$ git clone https://github.com/[your-name]/ScratchAddons.git
+```
+
+### Downloading the zipball
+
+If you don't have Git installed, you can try this method instead.
+
+1. Go to the [repository](https://github.com/ScratchAddons/ScratchAddons) and find the download code button
```
Comment (apple502j) [inline content/docs/getting-started/installing.md:41]: dot?

File: content/docs/getting-started/quick-start.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
+---
+title: Quick Start
+weight: 1
+aliases: 
+  - /docs/developing/installing-scratch-addons
+---
+
+## Installing
+
+Let's get started by installing Scratch Addons. Scratch Addons is available in both Chrome Web Store and Add-ons for Firefox.
```
Comment (apple502j) [inline content/docs/getting-started/quick-start.md:10]:
```suggestion
Let's get started by installing Scratch Addons. Scratch Addons is available in both Chrome Web Store and addons.mozilla.org.
```

File: content/docs/getting-started/quick-start.md [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+- [Chrome Web Store](https://chrome.google.com/webstore/detail/fbeffbjdlemaoicjdapfpikkikjoneco) (for Google Chrome and other Chromium-based browsers)  
+- [Add-ons for Firefox](https://addons.mozilla.org/firefox/addon/scratch-messaging-extension/) (for Mozilla Firefox)
```
Comment (apple502j) [inline content/docs/getting-started/quick-start.md:13]:
```suggestion
- [addons.mozilla.org](https://addons.mozilla.org/firefox/addon/scratch-messaging-extension/) (for Mozilla Firefox)  
```

File: content/docs/localization/_index.md [hunk 12]
```diff
@@ -1,18 +1,18 @@
...
-- [How to join the localization team](how-to-join-the-localization-team)
+- [Joining the Localization Team](joining-the-localization-team)
 - [Translating Scratch Addons](translating-scratch-addons) _Needs writing_
 
+## For developers
+- [Localizing Addons](localizing-addons)
+- [Localizing Background](localizing-background) _Needs writing_
+- [Technical specs of addons-l10n translation](https://github.com/ScratchAddons/ScratchAddons/blob/master/addons-l10n/README.md)
+
 ## For Crowdin admins
```
Comment (apple502j) [inline content/docs/localization/_index.md:17]:
```suggestion
## For Transifex admins
```

File: content/docs/reference/addon-api/_index.md [hunk 13]
```diff
@@ -0,0 +1,9 @@
+---
+title: Addon API
+description: This API can be used within your addons, which can provide useful things such as access to the real blockly, Scratch's vm, etc.
+aliases: 
+  - /docs/developing/addon-apis-reference
+---
+
+This API can be used within your addons, which can provide useful things such as access to the real blockly, Scratch's vm, etc.
```
Comment (apple502j) [inline content/docs/reference/addon-api/_index.md:8]:
```suggestion
This API can be used within your addons, which can provide useful things such as access to the real Blockly, Scratch's VM, etc.
```

File: content/docs/reference/addon-api/addon.badge.md [hunk 14]
```diff
@@ -0,0 +1,23 @@
+---
+title: addon.badge
+description: Allows addons to display a badge, a bit of text that is layered over the extension icon, frequently a number.
+weight: 7
+---
+
+**Keep in mind: as of August 31st, only one addon can use the badge API, currently `msg-count-badge`**
```
Comment (apple502j) [inline content/docs/reference/addon-api/addon.badge.md:7]:
```suggestion
**Keep in mind: only one addon (currently `msg-count-badge`) can use the badge API.**
```

File: content/docs/reference/addon-api/addon.notifications.md [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+**Required permissions: `notifications`**
+
+Allows addons to create browser notifications. This API slightly differs from the `browser.notifications` API extensions can use.
+
+## Functions
+### addon.notifications.create(optionsObject)
+Returns a promise that resolves to the ID (string) of the created notification.  
+Shows a notification to the user according to the options object.  
+[Options object reference](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/notifications/NotificationOptions).  
+Note: for your convenience, neither specifying `buttons` and `requireInteraction` throws on Firefox.
```
Comment (apple502j) [inline content/docs/reference/addon-api/addon.notifications.md:18]: Unclear, maybe this?
```suggestion
Note: for your convenience, specifying `buttons` or `requireInteraction` throws on Firefox.
```

File: content/docs/reference/addon-api/addon.settings.md [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+title: addon.settings
+description: Allows addons to change their behavior according to user-specified addon settings.  
+weight: 2
+---
+
+**Available in persistent scripts.**  
+**Available in userscripts.**
+
+Allows addons to change their behavior according to user-specified addon settings.  
+This API is available even if an addon doesn't specify `options` in its manifest, however all calls to `get()` will fail.
```
Comment (apple502j) [inline content/docs/reference/addon-api/addon.settings.md:11]:
```suggestion
This API is available even if an addon doesn't specify `settings` in its manifest, however all calls to `get()` will fail.
```

File: content/docs/reference/addon-api/addon.settings.md [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+
+## Functions
+### addon.settings.get(optionIdString)
+Returns the user-specified value for that option, or the default specified in the addon manifest if the user didn't specify a value by themselves.
```
Comment (apple502j) [inline content/docs/reference/addon-api/addon.settings.md:15]:
```suggestion
Returns the user-specified value for that setting, or the default specified in the addon manifest if the user didn't specify a value by themselves.
```

File: content/docs/reference/addon-api/addon.settings.md [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+**Available in persistent scripts.**  
+**Available in userscripts.**
+
+Allows addons to change their behavior according to user-specified addon settings.  
+This API is available even if an addon doesn't specify `options` in its manifest, however all calls to `get()` will fail.
+
+## Functions
+### addon.settings.get(optionIdString)
+Returns the user-specified value for that option, or the default specified in the addon manifest if the user didn't specify a value by themselves.  
+Throws if the specified option ID wasn't declared inside `addon.json`.
```
Comment (apple502j) [inline content/docs/reference/addon-api/addon.settings.md:16]:
```suggestion
Throws if the specified setting ID wasn't declared inside `addon.json`.
```

File: content/docs/reference/addon-api/addon.tab/_index.md [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+---
+
+**NOT available in persistent scripts.**  
+**Available in userscripts.**  
+
+Allows addon userscripts to get information about the tab they're currently running on.
+
+## Sub-APIs
+### [addon.tab.traps](addon.tab.traps)
+Allows addons to get direct references to objects, which are particularly useful for enhancing the editor, like the Scratch VM, Thread objects, or Redux state on scratch-www pages.
```
Comment (apple502j) [inline content/docs/reference/addon-api/addon.tab/_index.md:14]:
```suggestion
Allows addons to get direct references to objects, which are particularly useful for enhancing the editor, like the Scratch VM or the Blockly instance.
```

File: content/docs/reference/addon-api/addon.tab/_index.md [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+### [addon.tab.redux](addon.tab.redux)
+TODO
+
+## Getters
+### addon.tab.clientVersion
+Currently, the Scratch community website has 2 working clients used throughout the site, one React based and another jQuery based. This getter allows addons to change your behavior depending on the version of the current page.  
+Returns either `"scratch-www"` (React based), `"scratchr2"` (jQuery based) or `null`.
+### addon.tab.editorMode
+If the tab is a project, it returns the viewing mode of the project: `"editor"`, `"fullscreen"` or `"embed"`.
```
Comment (apple502j) [inline content/docs/reference/addon-api/addon.tab/_index.md:23]:
```suggestion
If the tab is a project, it returns the viewing mode of the project: `"projectpage"`, `"editor"`, `"fullscreen"` or `"embed"`.  
```

**Review by apple502j — CHANGES_REQUESTED** [review]

**Review by WorldLanguages — APPROVED** [review]
LGTM, apple502j suggested many good changes

**Comment by WorldLanguages** [comment]
One thing I could see being added is using a table for "available in userscripts / persistent scripts", instead of simple text. Whenever popup scripts are a thing, it could be added to that table

**Comment by Hans5958** [comment]
@apple502j

About the name of AMO, this is actually something that I consider for a while. I wanted to avoid naming it "addons.mozilla.org" to avoid confusion. I went with "Add-ons for Firefox" because that is the page title when I open https://addons.mozilla.org/en-US/firefox/, and [that is what Wikipedia went](https://en.wikipedia.org/wiki/Add-on_(Mozilla)#Website).

Let me know if this is fine.

File: content/docs/getting-started/installing.md (hunk 4 above)
Comment (Hans5958) [reply content/docs/getting-started/installing.md:37]: This, along with a heck ton of pics, probably could be solved later. For now it's fine.

**Comment by Hans5958** [comment]
If there are no more concerns that I may just merge this. Please leave mergings to me.

File: .gitignore [hunk 21]
```diff
@@ -1,5 +1,3 @@
 public/
 resources/_gen/
-hugo_stats.json
-# Local Netlify folder
-.netlify
\ No newline at end of file
+hugo_stats.jsongit push
```
Comment (GrahamSH-LLK) [inline .gitignore:3]: This was probably a mistake

**Pull request merged**